torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -357,15 +359,139 @@ def _init_from_local_shards(
         # add to metadata and local_shards
         sharded_tensor._metadata = global_sharded_tensor_metadata
         sharded_tensor._local_shards = local_shards
-        # make a EnumerableShardingSpec for sharded tensors that initialized from this API.
-        # TODO: make sharding spec a ChunkShardingSpec by inferring from the metadata list.
-        #       see issue https://github.com/pytorch/pytorch/issues/67244
-        sharded_tensor._sharding_spec = EnumerableShardingSpec(global_sharded_tensor_metadata.shards_metadata)
+        sharded_tensor._sharding_spec = _infer_sharding_spec_from_shards_metadata(
+            global_sharded_tensor_metadata.shards_metadata
+        )
 
         # run post initialization, i.e. map registration, rpc initialization
         sharded_tensor._post_init()
         return sharded_tensor
 
+    @classmethod
+    def _init_from_local_tensor(
+        cls,
+        local_tensor: torch.Tensor,
+        sharding_spec: ShardingSpec,
+        *global_size: Sequence[int],
+        process_group: dist.ProcessGroup = None,
+        init_rrefs=False,
+    ) -> "ShardedTensor":
+        """
+        Initialize a ShardedTensor with only one local tensor with global sharded tensor
+        size and sharding spec on each rank.
+
+        Args:
+            local_tensor (Tensor): Partial result stored in each rank.
```

Comment:
nit: local shard tensor stored in each rank? Partial results sounds a bit confusing

Reply:
Maybe let's call it local tensor from the local shard in each rank?